sql/src/main/java/io/crate/expression/scalar/timestamp/TimezoneFunction.java

```diff
@@ -0,0 +1,128 @@
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Locale;
+
+public class TimezoneFunction extends Scalar<Long, Object> {
+
+    public static final String NAME = "timezone";
+    private static final ZoneId UTC = ZoneId.of("UTC");
+    private static final FuncParams FUNCTION_PARAMS = FuncParams.builder(Param.STRING,
+                                                                         Param.of(DataTypes.TIMESTAMPZ,
+                                                                                  DataTypes.TIMESTAMP,
+                                                                                  DataTypes.LONG)).build();
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(NAME, new BaseFunctionResolver(FUNCTION_PARAMS) {
+            @Override
+            public FunctionImplementation getForTypes(List<DataType> signatureTypes) throws IllegalArgumentException {
+                DataType returnType = null;
+                int timestampParamTypeId = signatureTypes.get(1).id();
+                switch (timestampParamTypeId) {
+                    case LongType.ID:
+                    case TimestampType.ID_WITHOUT_TZ:
+                        returnType = DataTypes.TIMESTAMPZ;
+                        break;
+                    case UndefinedType.ID:
+                    case TimestampType.ID_WITH_TZ:
+                        returnType = DataTypes.TIMESTAMP;
+                        break;
+                    default:
+                        throw new IllegalArgumentException(String.format(
+                            Locale.ENGLISH,
+                            "invalid argument to %s: expected second parameter" +
+                            " to be of either type [%s, %s, %s] but found %s",
+                            NAME,
+                            LongType.INSTANCE.getName(),
+                            TimestampType.INSTANCE_WITH_TZ.getName(),
+                            TimestampType.INSTANCE_WITHOUT_TZ.getName(),
+                            signatureTypes.get(1)));
+                }
+                return new TimezoneFunction(new FunctionInfo(new FunctionIdent(NAME, signatureTypes), returnType));
+            }
+        });
+    }
+
+    private final FunctionInfo info;
+
+    private TimezoneFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public Long evaluate(TransactionContext txnCtx, Input<Object>... args) {
+        assert args.length == 2 : String.format(Locale.ENGLISH,
+                                                "number of arguments must be 2, got %d instead",
+                                                args.length);
+        String zoneStr = (String) args[0].value();
+        if (zoneStr == null) {
+            return null;
+        }
+        ZoneId zoneId = null;
```

Comment:
Same here, the assignment to `null` looks redundant.

Reply:
done